beets/dbcore/types.py

```diff
@@ -190,6 +190,38 @@ class String(Type):
     query = query.SubstringQuery
 
 
+class StringList(Type):
+    """A List of Unicode strings type, stored in a string
+    """
+    sql = u'TEXT'
+    query = query.SubstringQuery
+
+    def from_sql(self, sql_value):
+        return self.parse(sql_value)
+
+    def to_sql(self, value):
+        return self.format(value)
+
+    def format(self, value):
+        """Given a value of this type, produce a Unicode string
+        representing the value. This is used in template evaluation.
+        """
+
+        if isinstance(value, list):
+            value = ';'.join(value)
+
+        return six.text_type(value)
+
+    def parse(self, string):
+        """Parse a (possibly human-written) string and return the
+        indicated value of this type.
+        """
+        if string is None:
+            return self.null
+        else:
+            return string.split(';')
```

Comment:
We should probably make the value of list types like this a tuple rather than a Python list. That way, we avoid letting programmers mutate the list of genres. For example:

```
genres = item.genre
genres.append(foo)
item.store()
```

That wouldn't work, because the `Item` doesn't know you changed that copy of the list. We can capture this property by making the value immutable, i.e., a tuple.